tests/diagnostics/uninitialized-fields.slang

```diff
@@ -104,5 +104,28 @@ struct Impl
     }
 }
 
+// Calling a method from a constructor to initialize fields
```

Comment:
Put this into a separate test. `//CHECK-NOT` will succeed if the line isn't found from the last succesful match. If the message exists early before other `//CHECK` comments, this test will still suceed. Having this in its own test is the only certain way to make sure the warning doesn't exist.